### PR TITLE
Fixing Flaky Interactive Tests

### PR DESCRIPTION
**Pull request by ivanbasov**
Fixes:  #18283 and #18133 

Tagging @dotnet/roslyn-ide

### CODE REVIEW

**Comment by 333fred** [comment]
@ivanbasov something is still broken: https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_vs-integration_prtest/1917/testReport/junit/Roslyn.VisualStudio.IntegrationTests.CSharp/CSharpSendToInteractive/ResetInteractiveFromProjectAndVerify/

**Comment by ivanbasov** [comment]
Thank you, @333fred ! I will take a look.

**Comment by ivanbasov** [comment]
Could not find exact reason for the latest failure. Also could not reproduce it. 

Made the verification condition weaker but still enough. Added more diagnostics.

**Comment by tmat** [comment]
:shipit:

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpSendToInteractive.cs [hunk 1]
```diff
@@ -211,6 +213,7 @@ public void AddAssemblyReferenceAndTypesToInteractive()
             this.SubmitText("public class MyClass { public string MyFunc() { return \"MyClass.MyFunc()\"; } }");
             this.SubmitText("(new MyClass()).MyFunc()");
             this.WaitForLastReplOutput("\"MyClass.MyFunc()\"");
+            this.WaitForAsyncOperations(FeatureAttribute.SolutionCrawler);
```
Comment (jasonmalinowski) [inline src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpSendToInteractive.cs]: @ivanbasov Why was it necessary to have a wait at the end of the test? That seems surprising to me.

**Comment by ivanbasov** [comment]
Otherwise refreshing Visual Studio fails when switching to the next test. See #18133. It is easy to reproduce. Just comment out this.WaitForAsyncOperations(FeatureAttribute.SolutionCrawler); and try running all tests within CSharpSendToInteractive.cs. Also please see #18293.

**Pull request merged**